### PR TITLE
Aaronfay/ch4008/check jenkins plugin for breaking api change

### PR DESCRIPTION
**Pull request by aaronfay**
This update includes a few fixes:

 * add ability to handle an API change where multiple suite results can be returned in the response,
 * add the ability ([based on this PR](https://github.com/jenkinsci/ghost-inspector-plugin/pull/3)) to execute multiple suites at once as a comma-separated list of ids,
 * improves error handling for API calls and cleans up the plugin output,
 * refactors out the Ghost Inspector business objects and adds unit tests.

I have a little more work in getting the automated build cleaned up and passing (though it's close) but this should be ready for review now.

### CODE REVIEW

**Comment by aaronfay** [comment]
Here's an example of the new output formatting:

```
Started by user admin
Running as SYSTEM
Building in workspace /var/jenkins_home/workspace/standard-test

#####################################
Run Ghost Inspector Test Suite
#####################################

Configuration:
 - id(s): 5bb6451aa9f228640376c021, 604be5977851d67ae42e1c3b
 - startUrl: https://ghostinspector.com
 - params: paramOne=1234&jobName=standard-test

Executing:
 - executing URL: https://api.ghostinspector.com/v1/suites/5bb6451aa9f228640376c021/execute?immediate=1&startUrl=https%3A%2F%2Fghostinspector.com&paramOne=1234&jobName=standard-test&apiKey=9575xxx
 - suite [5bb6451aa9f228640376c021] returned 1 results:
  -> [604c073e55f33f4734288b1b]
 - executing URL: https://api.ghostinspector.com/v1/suites/604be5977851d67ae42e1c3b/execute?immediate=1&startUrl=https%3A%2F%2Fghostinspector.com&paramOne=1234&jobName=standard-test&apiKey=9575xxx
 - suite [604be5977851d67ae42e1c3b] returned 3 results:
  -> [604c073e95b9691125d620c3]
  -> [604c073e95b9691125d620c4]
  -> [604c073e95b9691125d620c5]
Checking results:
 - result [604c073e55f33f4734288b1b] status: passing null / failing: null
 - result [604c073e95b9691125d620c3] status: passing null / failing: null
 - result [604c073e95b9691125d620c4] status: passing null / failing: null
 - result [604c073e95b9691125d620c5] status: passing null / failing: null
Checking results:
 - result [604c073e55f33f4734288b1b] status: passing 1 / failing: 0
 ✓ result complete [604c073e55f33f4734288b1b] status: passing
 - result [604c073e95b9691125d620c3] status: passing null / failing: null
 - result [604c073e95b9691125d620c4] status: passing null / failing: null
 - result [604c073e95b9691125d620c5] status: passing null / failing: null
Checking results:
 - result [604c073e95b9691125d620c3] status: passing null / failing: null
 - result [604c073e95b9691125d620c4] status: passing null / failing: null
 - result [604c073e95b9691125d620c5] status: passing null / failing: null
Checking results:
 - result [604c073e95b9691125d620c3] status: passing null / failing: null
 - result [604c073e95b9691125d620c4] status: passing null / failing: null
 - result [604c073e95b9691125d620c5] status: passing null / failing: null
Checking results:
 - result [604c073e95b9691125d620c3] status: passing 1 / failing: 0
 ✓ result complete [604c073e95b9691125d620c3] status: passing
 - result [604c073e95b9691125d620c4] status: passing 1 / failing: 0
 ✓ result complete [604c073e95b9691125d620c4] status: passing
 - result [604c073e95b9691125d620c5] status: passing 1 / failing: 0
 ✓ result complete [604c073e95b9691125d620c5] status: passing
Finished: SUCCESS
```

File: pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
   <groupId>com.ghost-inspector.jenkins</groupId>
   <artifactId>ghost-inspector</artifactId>
-  <version>3.2-SNAPSHOT</version>
+  <version>3.3</version>
```
Comment (justinklemm) [inline pom.xml:14]: I can't remember offhand, but I think the plugin publishing process might increment this for us... I'll check.

**Review by justinklemm — COMMENTED** [review]
This looks really good. Thanks for all the hard (likely painful) work, and the tests! 🙏

If you haven't already, we'll also want to boot up a copy of Jenkins and manual test this just because it's such a big change. There are Docker contains that make this pretty simple.

**Comment by aaronfay** [comment]
Yeah, this is fully tested against Jenkins manually and also in our CircleCI config, we have a few steps that build the plugin, spin up a Jenkins docker container and run a [Ghost Inspector suite](https://app.ghostinspector.com/suites/5bb6451aa9f228640376c021) against the new plugin build to verify. I need to adjust the build to account for the new [data sources](https://ghostinspector.com/docs/data-driven-testing/#execute-from-data-source) API change, but after that we should be ready to merge this down. I'll get these builds cleaned up and ping you again when I have everything ready.

**Comment by justinklemm** [comment]
🙏 Awesome.

**Comment by aaronfay** [comment]
@justinklemm this should be good to go now, I've updated the CircleCI environment to test two suites, [one with a single test](https://app.ghostinspector.com/suites/5bb6451aa9f228640376c021) and [one with a data source](https://app.ghostinspector.com/suites/604be5977851d67ae42e1c3b) which both get executed at the same time using the new suite syntax `suiteId, suiteId`.

I'm not sure if you want to build this again yourself, or if you want you can pull [the `.hpi` file off the build if you want](https://app.circleci.com/pipelines/github/jenkinsci/ghost-inspector-plugin/29/workflows/e8bb5eb7-a96e-4000-a16d-8765f392dbce/jobs/71/artifacts).

**Review by justinklemm — APPROVED** [review]
LGTM! Let's merge this and I'll do the build/submission process tomorrow.

**Pull request merged**